source/testing/index.rst

```diff
@@ -0,0 +1,499 @@
+
+Discovery and Running
+~~~~~~~~~~~~~~~~~~~~~
+
+Both Django and Python’s standard library include test runners, and
+there are numerous others available from the community. The most
+popular third party runner is probably Nose, which also has a
+Django-specific extension, django_nose.
+
+Regardless of the runner you use, an increasingly popular step is to
+run all your tests with the tox utility, which allows you to define a
+number of environments you want to test and will run your test suite
+across all of them. This lets you ensure a code base works correctly
+not only for your local Python and Django version, for example, but
+across multiple versions and combinations of versions you want to
+support. Even for internal projects where you control the versions
+deployed to, this can be invaluable when preparing for dependency
+upgrades, such as migrating to a new Django release.
+
+References:
+
+* `unittest — Unit testing framework <http://docs.python.org/2/library/unittest.html>`_
+* `Test Discovery <http://docs.python.org/2/library/unittest.html#test-discovery>`_
+* `Nose <https://nose.readthedocs.org/en/latest/>`_
+* `django-nose <https://github.com/jbalogh/django-nose>`_
+
+Continuous Integration
+~~~~~~~~~~~~~~~~~~~~~~
+
+Continuous Integration (CI) is a term we mis-use to refer to
+automatically running our test suite on our project whenever
+there are changes.
+
+See http://caktus.github.io/developer-documentation/services/travis.html
+for help using Travis CI with Caktus Django projects.
+
+References:
+
+* `unittest-xml-reporting <https://pypi.python.org/pypi/unittest-xml-reporting/>`_
+
+Coverage
+--------
+
+With either an existing code base or a new project, you want to be
+sure as much of your code as possible is being tested. Rather than
+guessing, the coverage.py tool makes tracking test coverage easy. It
+runs any command, usually your test runner, and creates a report of
+every line of code executed, and the percentage of each module that
+was actually run during your test execution. You can monitor your
+project's test coverage, and make sure added new code doesn’t reduce
+your coverage. A good strategy with existing code bases is to find
+your current coverage rating and set this as a requirement for any new
+code -- no change is allowed which reduces test coverage. Over time,
+you'll your project’s code coverage to a reasonable level.
+
+As one example of how to use this, the `django_nose` project includes
+simple settings in your Django project to enable running all your
+tests through coverage and generating reports for you automatically::
+
+    INSTALLED_APPS = (
+        # ...
+        'django_nose',
+    )
+
+    # Use nose to run all tests
+    TEST_RUNNER = 'django_nose.NoseTestSuiteRunner'
+
+    # Tell nose to measure coverage on the 'foo' and 'bar' apps
+    NOSE_ARGS = [
+        '--with-coverage',
+        '--cover-package=foo,bar',
+    ]
+
+Another approach is to use a script to run your tests::
```

Comment:
I'd be in favor of just showing this example and ignoring the django-nose one. Personally I haven't seen much benefit to django-nose, and we haven't used it in recent projects. JMHO though.